mllib/src/main/scala/org/apache/spark/ml/param/params.scala

```diff
@@ -166,6 +166,18 @@ object ParamValidators {
   def inArray[T](allowed: java.util.List[T]): T => Boolean = { (value: T) =>
     allowed.contains(value)
   }
+
+  /** Check that the array length is greater than lowerBound. */
+  def arrayLengthGt[T](lowerBound: Double): T => Boolean = { (value: T) =>
```

Comment:
Change `T` to `Array[T]` directly and simply the impl:

~~~
def arrayLengthGt[T](lowerBound: Double): Array[T] => Boolean = { (value: Array[T]) =>
  require(value.length > lowerBound, "...")
}
~~~